project_stock/models/project_task.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from odoo import _, api, fields, models
+
+
+class Task(models.Model):
+    _inherit = "project.task"
+
+    @api.model
+    def _get_default_picking_type(self):
+        picking_type = False
+        if self.env.context.get("default_project_id"):
+            project = self.env["project.project"].browse(
+                self.env.context.get("default_project_id")
+            )
+            picking_type = project.picking_type_id
+        if not picking_type:
+            picking_type = self.env["stock.picking.type"].search(
+                [
+                    ("code", "=", "ptm_operation"),
+                    ("warehouse_id.company_id", "=", self.env.company.id),
+                ],
+                limit=1,
+            )
+        return picking_type.id if picking_type else False
+
+    @api.model
+    def _get_default_location_src_id(self):
+        location = False
+        if self.env.context.get("default_project_id"):
+            project = self.env["project.project"].browse(
+                self.env.context.get("default_project_id")
+            )
+            location = project.location_src_id
+        if not location:
+            warehouse = self.env["stock.warehouse"].search(
+                [("company_id", "=", self.env.company.id)], limit=1
+            )
+            location = warehouse.lot_stock_id
+        return location.id if location else False
+
+    @api.model
+    def _get_default_location_dest_id(self):
+        location = False
+        if self.env.context.get("default_project_id"):
+            project = self.env["project.project"].browse(
+                self.env.context.get("default_project_id")
+            )
+            location = project.location_dest_id
+        if not location:
+            warehouse = self.env["stock.warehouse"].search(
+                [("company_id", "=", self.env.company.id)], limit=1
+            )
+            location = warehouse.lot_stock_id
+        return location.id if location else False
+
+    procurement_group_id = fields.Many2one(
+        comodel_name="procurement.group", string="Procurement Group", copy=False
+    )
+    scrap_ids = fields.One2many(
+        comodel_name="stock.scrap", inverse_name="task_id", string="Scraps"
+    )
+    scrap_count = fields.Integer(
+        compute="_compute_scrap_move_count", string="Scrap Move"
+    )
+    move_raw_ids = fields.One2many(
```

Comment:
Bad name. This is not a production order with raw and finished products.

```suggestion
    move_ids = fields.One2many(
```